### PR TITLE
[C++ Worker]Fix for mac

### LINKED ISSUE #17537: [C++ API] Support loading .so files on macOS

### PR DESCRIPTION
**Pull request by qicosmos**
## Why are these changes needed?

Fix a gflags problem on mac: 

ERROR: something wrong with flag 'ray_address' in file 'cpp/src/ray/config_internal.cc'.  One possibility: file 'cpp/src/ray/config_internal.cc' is being linked both statically and dynamically into this executable.

Replace gflags with absl::flags solve the probelem.

## Related issue number

Closes #17537 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests

### CODE REVIEW

**Comment by SongGuyang** [comment]
Can we also add a Mac CI test for C++ worker?

File: cpp/src/ray/runtime/object/native_object_store.cc [hunk 1]
```diff
@@ -65,14 +65,17 @@ std::shared_ptr<msgpack::sbuffer> NativeObjectStore::GetRaw(const ObjectID &obje
 
 void NativeObjectStore::CheckException(const std::string &meta_str,
                                        const std::shared_ptr<Buffer> &data_buffer) {
+  std::string data_str =
+      data_buffer ? std::string((char *)data_buffer->Data(), data_buffer->Size()) : "";
+
```
Comment (SongGuyang) [inline cpp/src/ray/runtime/object/native_object_store.cc:70]: Why modify this file in current PR?

**Review by SongGuyang — COMMENTED** [review]
Some comments.

**Comment by SongGuyang** [comment]
Please add an Mac CI in `.buildkite/pipeline.macos.yml`

**Review by SongGuyang — APPROVED** [review]
LGTM

**Review by raulchen — APPROVED** [review]

**Pull request merged**